### PR TITLE
feat(core): add spinner during node module install

### PR DESCRIPTION
**Pull request by juristr**
## Current Behavior

Right now there is no spinner, just a static message informing about multiple installs being done. If you're on a slow connection it looks like the window freezed

## Expected Behavior

Adds a spinner and message "Installing dependencies" during the npm package install

Screenshot 2021-05-07 at 13 02 33

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
This adds a spinner but it doesn't spin. I think the spinning happens on the next tick but in this case `execSync` blocks the call stack from emptying and going onto the event loop. We need an asynchronous `exec` here. There's an `execAndWait` and that one records logs so maybe use that one instead.

File: packages/create-nx-workspace/bin/create-nx-workspace.ts [hunk 1]
```diff
@@ -476,20 +492,12 @@ function createSandbox(packageManager: string) {
...
   try {
-    execSync(`${packageManager} install --silent`, {
-      cwd: tmpDir,
-      stdio: 'ignore',
-    });
+    await execAndWait(`${packageManager} install --silent`, tmpDir);
   } catch (_) {
-    // Install failed so run again without --silent
     try {
-      execSync(`${packageManager} install`, {
```
Comment (FrozenPandaz) [inline packages/create-nx-workspace/bin/create-nx-workspace.ts:486]: This was only to get the logs. Now that we use `execAndWait`, we do not need to try it again.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
For some reason, the spinner lingers now if Nx Cloud is added.

<details>
  <summary>After</summary>
    jason@pop-os ~/p/t/n/temp [1]> pnpx create-nx-workspace@latest --pm npm
✔ Install the following package: create-nx-workspace@latest? (Y/n) · true
Packages: +48
++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /home/jason/.pnpm-store/v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 48, reused 48, downloaded 0, added 48, done

/home/jason/.npm/_npx/68501/pnpm-global/5:
+ create-nx-workspace 12.99.0
✔ Workspace name (e.g., org name)     · new-aweflnanfl
✔ What to create in the new workspace · express
✔ Application name                    · app1
✔ Use Nx Cloud? (It's free and doesn't require registration.) · Yes

>  NX  Nx is creating your workspace.

  To make sure the command works reliably in all environments, and that the preset is applied correctly,
  Nx will run "npm install" several times. Please wait.

⠼ Creating your workspace
———————————————————————————————————————————————

UPDATE nx.json

>  NX   NOTE  Nx Cloud has been enabled

  Your workspace is currently public. Anybody with code access can view the workspace on nx.app. 
  You can connect the workspace to your Nx Cloud account at https://nx.app/orgs/workspace-setup?accessToken=NjI2MzI2ZDItNGVlZi00MTFkLTg3MmYtOWQ1ODM3ZGIwNzc3fHJlYWQtd3JpdGU=. (You can do this later.)


>  NX   SUCCESS  Nx has successfully created the workspace.

</details>

<details>
  <summary>Before</summary>
  jason@pop-os ~/p/t/n/temp [SIGINT]> pnpx create-nx-workspace@12.0.8 --pm npm
✔ Install the following package: create-nx-workspace@12.0.8? (Y/n) · true
Packages: +48
++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /home/jason/.pnpm-store/v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 48, reused 47, downloaded 1, added 48, done

/home/jason/.npm/_npx/69238/pnpm-global/5:
+ create-nx-workspace 12.0.8 (12.99.0 is available)
✔ Workspace name (e.g., org name)     · new-alnwelfan
✔ What to create in the new workspace · react
✔ Application name                    · app1
✔ Default stylesheet format           · css
✔ Use Nx Cloud? (It's free and doesn't require registration.) · Yes

>  NX  Nx is creating your workspace.

  To make sure the command works reliably in all environments, and that the preset is applied correctly,
  Nx will run "npm install" several times. Please wait.


>  NX   SUCCESS  Nx has successfully created the workspace.


———————————————————————————————————————————————

UPDATE nx.json

>  NX   NOTE  Nx Cloud has been enabled

  Your workspace is currently public. Anybody with code access can view the workspace on nx.app. 
  You can connect the workspace to your Nx Cloud account at https://nx.app/orgs/workspace-setup?accessToken=NmQ3MGU3MDItMzA4OS00NGI4LThhY2YtY2M3Zjc0ODI5M2Q3fHJlYWQtd3JpdGU=. (You can do this later.)


———————————————————————————————————————————————


>  NX   NOTE  First time using Nx? Check out this interactive Nx tutorial.

  https://nx.dev/react/tutorial/01-create-application
  
  Prefer watching videos? Check out this free Nx course on Egghead.io.
  https://egghead.io/playlists/scale-react-development-with-nx-4038
  </details>

**Comment by juristr** [comment]
@FrozenPandaz rebased and fixed merge conflicts. should be ready now

image

**Review by FrozenPandaz — APPROVED** [review]
Beautiful :smile:

**Pull request merged**